src/openrct2/ride/Track.cpp

```diff
@@ -1491,9 +1491,10 @@ static money32 track_place(sint32 rideIndex,
         tileElement->type = map_type;
 
         tile_element_set_track_sequence(tileElement, trackBlock->index);
-        tileElement->properties.track.ride_index = rideIndex;
-        tileElement->properties.track.type       = type;
-        tileElement->properties.track.colour     = 0;
+        track_element_set_ride_index(tileElement, rideIndex);
+        track_element_set_type(tileElement, type);
+        // tileElement->properties.track.colour used to be set to 0 here, which is odd, considering it is memset to 0
```

Comment:
I would hardly call it odd that CS might forget that element has zeroed out this memory. This comment does not really add anything to the code base. We have proven at that it is zero and it does not need to be looked at. Have you considered that possibly this is set to zero due to using a `mov16` of `rideIndex` (Its not i checked).

Reply:
> Have you considered that possibly this is set to zero due to using a mov16 of rideIndex (Its not i checked).

Not sure what you mean with this. 

I'll take out the comment, is that correct?

Reply:
Yeah just remove the comment.

My bit about mov16 was just that it might have been a 16bit move of an 8 bit value filling the rest of the bits with zero (zero extend). I checked the assembly and this was not the case but that can at times be the reason for odd code like this.